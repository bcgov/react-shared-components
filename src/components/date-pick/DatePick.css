.bcgov-flex {
  display: flex;
}

.bcgov-move-right {
  margin-right: 5px;
}
