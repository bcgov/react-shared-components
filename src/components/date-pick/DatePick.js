import React from "react";
import PropTypes from "prop-types";
import DatePicker from "react-datepicker";
import { MdDateRange } from "react-icons/md";

import "./DatePick.css";
import "react-datepicker/dist/react-datepicker.css";

export const DatePick = ({ selectedDate, setSelectedDate }) => {
  return (
    <div className="bcgov-flex">
      <MdDateRange size={32} className="bcgov-move-right" />
      <DatePicker selected={selectedDate} onChange={setSelectedDate} />
    </div>
  );
};

DatePick.propTypes = {
  selectedDate: PropTypes.instanceOf(Date).isRequired,
  setSelectedDate: PropTypes.func.isRequired,
};
