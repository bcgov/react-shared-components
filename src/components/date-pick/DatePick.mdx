import { Meta, Story } from "@storybook/addon-docs";
import { DatePick } from "./DatePick";

<Meta title="DatePick" component={DatePick} />

# Date Pick

Date picker component for allowing users to select a date.

## API

| Prop            | Type     | Required | Description                                                  | Default |
| --------------- | -------- | -------- | ------------------------------------------------------------ | ------- |
| selectedDate    | date     | true     | sets the default selected date                               |         |
| setSelectedDate | function | true     | function to set the newly selected date state                |         |
| label           | string   | false    | label text to display above the date picker                  | ""      |
| isRequired      | boolean  | false    | determines whether date picker selection is mandatory or not | false   |

<br />

## Default Date Picker

Used for selecting a date.

<Story name="Default" id="date-pick--default"></Story>

```bash
const [selectedDate, setSelectedDate] = useState(new Date('August 19, 1975 23:15:30'));

<DatePick
  selectedDate={selectedDate}
  setSelectedDate={setSelectedDate}
/>
```

## With Label (Mandatory) Date Picker

Used when selecting a date is mandatory.

<Story name="With Label (Mandatory)" id="date-pick--with-label"></Story>

```bash
<DatePick
  label="Select date"
  isRequired
  selectedDate={selectedDate}
  setSelectedDate={setSelectedDate}
/>
```
