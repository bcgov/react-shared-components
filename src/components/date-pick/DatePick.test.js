import React from "react";

import testBasicSnapshot from "../../TestHelper";
import { DatePick } from "./DatePick";

describe("DatePick Component", () => {
  const selectedDate = new Date();
  const setSelectedDate = jest.fn();

  test("Matches the snapshot", () => {
    testBasicSnapshot(
      <DatePick selectedDate={selectedDate} setSelectedDate={setSelectedDate} />
    );
  });
});
