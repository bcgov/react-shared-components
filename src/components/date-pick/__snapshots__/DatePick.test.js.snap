// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePick Component Matches the snapshot 1`] = `
Array [
  "",
  <div
    className="bcgov-flex"
  >
    <svg
      className="bcgov-move-right"
      fill="currentColor"
      height={32}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20a2 2 0 002 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
      />
    </svg>
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <input
          className=""
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="text"
          value="08/19/1975"
        />
      </div>
    </div>
  </div>,
]
`;

exports[`DatePick Component With mandatory label matches the snapshot 1`] = `
Array [
  <div
    className="bcgov-input-label"
  >
    <label
      htmlFor="Select date"
    >
      Select date
    </label>
    <span
      className="bcgov-mandatory"
      id="asterisk"
    >
      *
    </span>
    Â 
  </div>,
  <div
    className="bcgov-flex"
  >
    <svg
      className="bcgov-move-right"
      fill="currentColor"
      height={32}
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width={32}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 0h24v24H0z"
        fill="none"
      />
      <path
        d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20a2 2 0 002 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
      />
    </svg>
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <input
          className=""
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          readOnly={false}
          type="text"
          value="08/19/1975"
        />
      </div>
    </div>
  </div>,
]
`;
